src/Api/PubnubApi/Security/Crypto/CryptoModule.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using PubnubApi.Security.Crypto.Cryptors;
+
+namespace PubnubApi.Security.Crypto
+{
+    public class CryptoModule
+    {
+        private readonly ICryptor _cryptor;
+        private readonly LegacyCryptor _fallbackCryptor;
+        public CryptoModule(ICryptor cryptor, LegacyCryptor fallbackCryptor)
+        {
+            _cryptor = cryptor;
+            _fallbackCryptor = fallbackCryptor;
+        }
+        public static LegacyCryptor CreateLegacyCryptor(string cipherKey, bool useDynamicIV, IPubnubLog log, IPubnubUnitTest unitTest)
+        {
+            return new LegacyCryptor(cipherKey, useDynamicIV, log, unitTest);
+        }
+        public static LegacyCryptor CreateLegacyCryptor(string cipherKey, bool useDynamicIV)
+        {
+            return new LegacyCryptor(cipherKey, useDynamicIV);
+        }
+        public static LegacyCryptor CreateLegacyCryptor(string cipherKey)
+        {
+            return new LegacyCryptor(cipherKey, true);
+        }
+        public static AesCbcCryptor CreateAesCbcCryptor(string cipherKey, IPubnubLog log, IPubnubUnitTest unitTest)
```

Comment:
Let's not pass UnitTests in constructors - we can invert it by exposing event delegates

Reply:
the only purpose is pass known IV for encryption, so that we can verify the expected output. Any other option?

Reply:
removed UnitTests from constructors.